src/catkin_pkg/packages.py

```diff
@@ -129,25 +129,27 @@ def find_packages_allowing_duplicates(basepath, exclude_paths=None, exclude_subs
     if not data:
         return {}
 
-    if len(data) < 100:
+    try:
+        if len(data) < 100:
+            raise Error
+        parser = _PackageParser(warnings is not None)
+        pool = multiprocessing.Pool()
+        try:
+            path_parsed_packages, warnings_lists = zip(*pool.map(parser, data))
+        finally:
+            pool.close()
+            pool.join()
+        if parser.capture_warnings:
+            map(warnings.extend, warnings_lists)
+        return dict(path_parsed_packages)
+    except:
```

Comment:
Try except the whole code block seems to be excessive as well as excepting on any kind of exception. Please identify a specific function which can be tested for to choose between the two approaches.